docs/components/componentdoc.vue

```diff
@@ -1,106 +1,148 @@
 <template>
   <section
+    v-if="component"
     class="bd-content"
-    v-if="component">
-
+  >
     <b-row
       tag="header"
-      align-v="center">
-      <b-col sm="9"><h2 :id="`comp-ref-${componentName}`"><code>{{ tag }}</code></h2></b-col>
+      align-v="center"
+    >
+      <b-col sm="9">
+        <h2 :id="`comp-ref-${componentName}`">
+          <code>{{ tag }}</code>
+        </h2>
+      </b-col>
       <b-col
         sm="3"
-        class="text-sm-right">
+        class="text-sm-right"
+      >
         <b-btn
           variant="outline-secondary"
           size="sm"
           :href="githubURL"
-          target="_blank">view source</b-btn>
+          target="_blank"
+        >
+          view source
+        </b-btn>
       </b-col>
     </b-row>
 
     <article v-if="aliases && aliases.length > 0">
-      <h4 :id="`comp-ref-${componentName}-aliases`">Component aliases</h4>
-      <p><code>{{ tag }}</code> can also be used via the following aliases:</p>
+      <h4 :id="`comp-ref-${componentName}-aliases`">
+        Component aliases
+      </h4>
+      <p>
+        <code>{{ tag }}</code> can also be used via the following aliases:
+      </p>
       <ul>
-        <li v-for="alias in aliases" :key="alias"><code>&lt;{{ kebabCase(alias) }}&gt;</code></li>
+        <li
+          v-for="alias in aliases"
+          :key="alias"
+        >
+          <code>&lt;{{ kebabCase(alias) }}&gt;</code>
+        </li>
       </ul>
     </article>
 
     <article v-if="props_items && props_items.length > 0">
-      <h4 :id="`comp-ref-${componentName}-props`">Properties</h4>
+      <h4 :id="`comp-ref-${componentName}-props`">
+        Properties
+      </h4>
```

Comment:
The only thing about this, is that all that whitespace (inside the `<h4>`) is stored in the compiled render function.

Reply:
I think this is related to the [`vue/singleline-html-element-content-newline`](https://eslint.vuejs.org/rules/singleline-html-element-content-newline.html) ESlint rule.

Reply:
Might be.  Things like inline elements (`<span>`, `<strong>`, etc) should all be allowed to be inlined without a line break, to reduce final render function size (unless tehre is a rule to strip whitespace, but then that can affect things like `<pre>` etc.

The following will not render the same and will have slightly different spacing on screen:

```html
<span><input type="checkbox"><label>Foobar</label></span>
```

vs.

```html
<span>
  <input type="checkbox">
  <label>Foobar</label>
</span>
```

The latter will introduce spacing between the label and the input (via whitespace), while the former will have night tight spacing.  Inspecting the rendered output in chrome dev tools will show the extra whitesapce.

![image](https://user-images.githubusercontent.com/2781561/51076789-ba0d6180-1673-11e9-96fc-6f149ec871f5.png)

So I don't always agree to breaking tags across lines, specifically when one doesn't want to introduce side effects of adding spaces between tags.  There may be times where we want to have no spaces between the tags (images and outer containers is usually a big one where the white space affects image alignment).

Is there a way to override this?

Reply:
The other issue is in the live examples in the README docs.  We try to reduce the "height" of the code (i.e. not breaking some tags across multiple lines), so that the user doesn't have miles to scroll on the screen to get to the next section of docs.  If an example can be written on one line, it should stay that way (i.e. `<b-button @click="foo">Bar</b-button>` should remain as a single line and not broken across three lines).